File: backend/app/schemas/event_asset_link.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class EventAssetLinkResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    event_id: UUID
    asset_id: UUID
    symbol: str | None = None
    name: str | None = None
    impact_direction: str
    impact_strength: float
    reason: str | None
    confidence_score: float | None
    created_at: datetime

File: frontend/src/types/eventAssetLink.ts
export interface EventAssetLink {
  id: string;
  event_id: string;
  asset_id: string;
  symbol?: string;
  name?: string;
  impact_direction: string;
  impact_strength: number;
  reason: string;
  confidence_score: number;
  created_at: string;
}